codec/codec-fury/src/main/java/com/alipay/sofa/rpc/codec/fury/FurySerializer.java

```diff
@@ -0,0 +1,310 @@
+                for (String key : userMap.keySet()) {
+                    checker.allowClass(key);
+                }
+                return f;
+            });
+
+        } else {
+            fury = new ThreadLocalFury(classLoader -> {
+
+                Fury f = Fury.builder().withLanguage(Language.JAVA)
+                    .withRefTracking(false)
+                    .withCodegen(true)
+                    .withNumberCompressed(true)
+                    .withCompatibleMode(COMPATIBLE)
+                    .requireClassRegistration(false)
+                    .withClassLoader(classLoader)
+                    .withAsyncCompilation(true)
+                    .build();
+                return f;
+            });
+        }
+
+    }
+
+    @Override
+    public AbstractByteBuf encode(final Object object, final Map<String, String> context) throws SofaRpcException {
+        if (object == null) {
+            throw buildSerializeError("Unsupported null message!");
+        } else if (object instanceof SofaRequest) {
+            return encodeSofaRequest((SofaRequest) object, context);
+        } else if (object instanceof SofaResponse) {
+            return encodeSofaResponse((SofaResponse) object, context);
+        } else {
+            return new ByteArrayWrapperByteBuf(fury.serialize(object));
+        }
+    }
+
+    private AbstractByteBuf encodeSofaRequest(SofaRequest sofaRequest, Map<String, String> context)
+        throws SofaRpcException {
+        Object[] args = sofaRequest.getMethodArgs();
+        if (args.length == 1) {
+            return encode(args[0], context);
+        } else {
+            return encode(args, context);
+        }
+    }
+
+    private AbstractByteBuf encodeSofaResponse(SofaResponse sofaResponse, Map<String, String> context)
+        throws SofaRpcException {
+        AbstractByteBuf byteBuf;
+        if (sofaResponse.isError()) {
+            byteBuf = encode(sofaResponse.getErrorMsg(), context);
+        } else {
+            Object appResponse = sofaResponse.getAppResponse();
+            if (appResponse instanceof Throwable) {
+                byteBuf = encode(((Throwable) appResponse).getMessage(), context);
+            } else {
+                byteBuf = encode(appResponse, context);
+            }
+        }
+        return byteBuf;
+    }
+
+    @Override
+    public Object decode(final AbstractByteBuf data, final Class clazz, final Map<String, String> context)
+        throws SofaRpcException {
+        Object result = null;
+        if (clazz == null) {
+            throw buildDeserializeError("class is null!");
+        } else if (data.readableBytes() <= 0) {
+            try {
+                result = clazz.newInstance();
+            } catch (InstantiationException | IllegalAccessException e) {
+                throw buildDeserializeError(e.getMessage());
+            }
+            return result;
+        } else {
+            result = fury.deserialize(data.array());
```

Comment:
` buffer.arrayOffset() + buffer.position()` should be 0 in this case, do we need a check?